### PR TITLE
Add complex logarithm to math

### PR DESCRIPTION
**Pull request by BeyondEspresso**
This brief pull request adds complex logarithms.

### CODE REVIEW

**Comment by BeyondEspresso** [comment]
Originally, I required this addition also for #2161 , but now this has become a standalone PR.

File: include/pmacc/math/complex/Complex.tpp [hunk 1]
```diff
@@ -180,6 +180,24 @@ struct Abs2< ::pmacc::math::Complex<T_Type> >
     }
 };
 
+/*  Specialize log() for complex numbers. */
+template< typename T_Type >
+struct Log< ::pmacc::math::Complex< T_Type > >
+{
+    typedef typename ::pmacc::math::Complex< T_Type >::type result;
+    typedef T_Type type;
```
Comment (ax3l) [inline include/pmacc/math/complex/Complex.tpp]:
```C++
using type = T_Type;
```

File: include/pmacc/math/complex/Complex.tpp [hunk 2]
```diff
@@ -180,6 +180,24 @@ struct Abs2< ::pmacc::math::Complex<T_Type> >
...
+
+    HDINLINE result operator( )( ::pmacc::math::Complex<T_Type> const & other )
```
Comment (ax3l) [inline include/pmacc/math/complex/Complex.tpp]: spaces around `T_Type` please :)

File: include/pmacc/math/complex/Complex.tpp [hunk 3]
```diff
@@ -180,6 +180,24 @@ struct Abs2< ::pmacc::math::Complex<T_Type> >
...
+    {
+        return pmMath::log( pmMath::abs( other ) ) +
+        ::pmacc::math::Complex< T_Type >(
```
Comment (ax3l) [inline include/pmacc/math/complex/Complex.tpp]: pls indent all by one level (4 spaces)

**Comment by BeyondEspresso** [comment]
@ax3l I included your code style suggestions.

File: include/pmacc/math/complex/Complex.tpp [hunk 4]
```diff
@@ -180,6 +180,24 @@ struct Abs2< ::pmacc::math::Complex<T_Type> >
     }
 };
 
+    /*  Specialize log() for complex numbers. */
+    template< typename T_Type >
+    struct Log< ::pmacc::math::Complex< T_Type > >
+    {
+        typedef typename ::pmacc::math::Complex< T_Type >::type result;
```
Comment (ax3l) [inline include/pmacc/math/complex/Complex.tpp]: nooooo... that's also a `using`

File: include/pmacc/math/complex/Complex.tpp [hunk 5]
```diff
@@ -180,6 +180,24 @@ struct Abs2< ::pmacc::math::Complex<T_Type> >
...
+        using type = T_Type;
+
+        HDINLINE result operator( )( ::pmacc::math::Complex< T_Type > const & other )
+        {
+            return pmMath::log( pmMath::abs( other ) ) +
+            ::pmacc::math::Complex< T_Type >(
```
Comment (ax3l) [inline include/pmacc/math/complex/Complex.tpp]: you did not align this as annotated ;)

File: include/pmacc/math/complex/Complex.tpp (hunk 4 above)
Comment (BeyondEspresso) [inline include/pmacc/math/complex/Complex.tpp]: You did not write that ;-) .

File: include/pmacc/math/complex/Complex.tpp (hunk 5 above)
Comment (BeyondEspresso) [inline include/pmacc/math/complex/Complex.tpp]: I have as requested indented the entire operator function within its name space. What else do want to indent?

File: include/pmacc/math/complex/Complex.tpp (hunk 4 above)
Comment (ax3l) [inline include/pmacc/math/complex/Complex.tpp]: I thought you will guess so ;)

```C++
using type = T_Type;
using result = typename ::pmacc::math::Complex< type >::type;
```

File: include/pmacc/math/complex/Complex.tpp (hunk 5 above)
Comment (ax3l) [inline include/pmacc/math/complex/Complex.tpp]: uh, that was also missing.
other request: offline

**Comment by BeyondEspresso** [comment]
Included these suggestions.

**Pull request merged**